app/src/main/java/org/projectbuendia/client/ui/chart/ChartRenderer.java

```diff
@@ -158,11 +158,9 @@ public void render(Chart chart, Map<String, Obs> latestObservations,
         void addObservations(List<Obs> observations) {
             for (Obs obs : observations) {
                 if (obs == null) continue;
-                ObsPoint point = obs.getObsPoint();
-                if (point == null) continue;
```

Comment:
The problem here is that getObsPoint returns null if there is no data type.  The "order executed" observations have no data type, so this is skipped and the order executions do not get counted.  But this works fine if we use obs.time directly instead of converting to ObsPoint first.

I moved the null check of the ObsPoint to addObs.  It's better anyway to protect near the site of the actual call that is sensitive to a null value, i.e. the call to add().